packages/flutter_tools/lib/src/dart/pub.dart

```diff
@@ -387,6 +398,12 @@ class _DefaultPub implements Pub {
     if (pubSpecYaml.lastModifiedSync().isAfter(dotPackagesLastModified)) {
       return true;
     }
+
+    if (_toolStampFile != null &&
```

Comment:
I think the characters issues was getting triggered by this code I removed @justinmc . Normally we don't need to invalidate the packages when a dependency changes because the pubspec must have changed, but the flutter sdk is special and needs a bit of help.

I don't think we missed it because so much else triggers a pub get that in most cases you won't notice

Reply:
Ah nice, thanks for following up on that!  The [issue](https://github.com/flutter/flutter/issues/59645) is still getting the occasional comment from someone experiencing the build problem, so hopefully this PR will clear that up.